src/02-common-principles.md

```diff
@@ -644,6 +540,147 @@ for more information.
 }
 ```
 
+## The Inheritance Principle
+
+1.  Any metadata file (such as `.json`, `.bvec` or `.tsv`) MAY be defined at any directory level.
+
+1.  For a given data file, any metadata file is applicable to that data file if:
+    1.  It is stored at the same directory level or higher;
+    1.  The metadata and the data filenames possess the same suffix;
+    1.  The metadata filename does not include any entity absent from the data filename.
+
+1.  A metadata file MUST NOT have a file name that would be otherwise applicable
+    to some data file based on rules 2.2 and 2.3 but is made inapplicable based on its
+    location in the directory structure as per rule 2.1.
+
+1.  There MUST NOT be multiple metadata files applicable to a data file at one level
+    of the directory hierarchy.
```

Comment:
FWIW: it is such a nice concise rule which helps to avoid ambiguity and the workaround in Example 3 is quite cute ;)

Reply:
OH -- I remembered what bothered me in my original #102 and outlined in Edit 1 there. Citing an example from there:

> I placed myself into a corner with an example of having e.g.
> 
> ```
> sub-1_task-task1_run-1_bold.json and
> sub-1_task-task1_acq-X_run-1_bold.json
> ```
>
> per subject (should be ok), and then trying to aggregate over them while retaining also _acq- if defined. 

and there are many other entities/use-cases which would fall into similar situation.  What I am thinking is to add a clarification as subitem here or a separate rule:

- A metadata file at the data files level of hierarchy MUST not be considered for inheritance if there is a matching in entities data file.

This would make `sub-1_task-task1_run-1_bold.json` not a to be considered for providing metadata to enrich `sub-1_task-task1_acq-X_run-1_bold.json`, and examples below would stay valid (we might want to add an example for this rule though)

Reply:
> it is such a nice concise rule which helps to avoid ambiguity and the workaround in Example 3 is quite cute ;)

Well, actually (from first post):

> Full disclosure: as per #259, I would like to pursue the prospect of modifying the inheritance principle itself, not just the description of such; specifically removal of the preclusion of having multiple applicable JSONs at one level of the hierarchy.

😬 

The proposed changes here do at least I think demonstrate what would need to change in order to facilitate such. Any text I propose for such will need to be very clear for both users and software creators exactly what is permitted vs. not permitted and in what order JSONs should be loaded. But as I said in the original post, this is not a change in descriptive language but a change in prescriptive permissible structure, and so IMO necessitates either a minor or major version change rather than just a patch.

Reply:
Gotcha. Given the currently loose formulation of the principle, depending on the changes to it, I think it might be feasible to get it done within minor revision. So it would be great to see this PR be finalized/merged soon.